Project-1/README.md

```diff
@@ -148,7 +148,10 @@ There exists a parent relationship between z and children u and w. u does not eq
 
 4. Let V be a vocabulary of the attribute (concept) language with complements (ALC) consisting of a role name "parent_of" and a concept name "Male". Interpret parent_of as "x is a parent of y" and M as "x is male". Where possible define the following formulas in this vocabulary; where not possible, explain why: 
   ```(a)  B that says that x is a brother of y
- 
+ person1 ≡ 
```

Comment:
Hi Olivia, for some reason I can only comment in this line. My thoughts on the overall thing:
1 looks good
2 a is good. 2 b and c are both tautologies, so you can just write A∨¬A or something like that
3 I don't think you should use the same Property you are trying to define inside the definition. Here is the definitions we came up during the last meeting:
B(x,y) that says that x is a brother of y  
  B(x,y) := ∃z.(P(zx) ∧ P(zy) ∧ ¬(z=y)) ∧ ¬(z=x) ¬(x=y) ¬F(x)
  (b)  A(x,y) that says that x is an aunt of y  
  A(x,y) := ∃z∃w.(P(zy) ∧ P(wz) ∧ P(wx) ∧ ¬(z=x)) F(x)
  (c)  C(x,y) that says that x and y are cousins   
  C(x,y) := ∃u∃w∃z(P(w,z)∧P(w,u)∧P(z,x)∧P(u,y)∧¬(z=u))∧¬(x=y)
  (d)  O(x) that says that x is an only child  
  O(x) := ∃y.(P(yx) ∧ ∀z.(P(yz) → (w=y)))
  (e)  T(x) that says that x has exactly two brothers 
  T(x) := ∃u∃y∃z(¬(y=z)∧¬(x=y)∧¬(x=z)∧P(u,x)∧P(u,y)∧P(u,z)∧¬F(y)∧¬F(z))∧∀w(P(u,w)∧¬F(w)→ w=x ∨ w=y ∨ w=z)

4 and 5 we did together in the meeting. I am attaching the results we got to.
(a)  B that says that x is a brother of y

  Per (Person) ≡ M ⊔ ¬M
  p2 (Parent of 2) ≡ ∃parentof. ≥ 2 ⊓ ¬(∃parentof. = 1) (this condition is meant to avoid that the two persons are the same, but I am not entirely sure it works or whether there are better ways to do so)
  
  B ≡ M ⊓ ∃p2¯.Per

  (b)  A that says that x is an aunt of y
  
  ∃p2¯.parentof.parentof.Per

  A ≡ ¬M ⊓ (∃p2¯. (parentof. Per ≥ 2).(parentof.Per)) ⊔ ¬parentof.Per 
  
  U(uncle) ≡ M ⊓ (∃p2¯. (parentof. Per ≥ 2).(parentof.Per)) ⊔ ¬parentof.Per 

  (c)  C that says that x and y are cousins

  gp2(Grandparent with at least two children which each have children) ≡ ∃parentof ≥ 2.parentof.Per

  C ≡ gp2¯.per

  But this would only describe one of the two cousins, i.e. "being a cousin of someone" or "having a cousin".
  We cannot define "being cousins" in description logic.

  (d)  O that says that x is an only child  

  O ≡ ¬ ∃p2¯.Per

  (e)  T that says that x has exactly two brothers 

  This cannot be done without a role name for brother, we only have a concept name for it.

  ```


5. Select two formulas defined in ALC from question 4 to form the basis of a T-Box. Supplement this T-box with whatever other axioms you like, as well as an A-box, so that you ultimately construct a knowledge base K = (T,A). Provide a _model_ of K. This may be graphical or symbolic or both. 

TBox = {
B ⊑ M
B ⊑ Per
M ⊑ Per
A ⊑ ¬M
A ⊑ Per
}

ABox = { John : M
John : parentof. Chris
John: B
Chris: M
Mary : ¬M
Mary: A
Alexander : parentof. Mary
Alexander : parentof. John
}

ΔI = {John, Mary, Alexander, Chris}

Named Individuals:
JohnI = J,
MaryI = M,
AlexanderI = A,
Chris = C

Concept Assignments:
PersonI = {J, M, A, C},
MaleI = {J, A, C},
FemaleI = {M},
BrotherI = {J},
AuntI = {M}

Role Assignments:
parentofI = {(J, C), (A, J), (A, M)}

6 looks good, you are missing one though right?

7 Here is what I did

{(a,b)(b,a)(a,a)} 
 The second statement is true in this model. There is an x that relates every y (a, related to a and b) There is an x that relates to a z (where a is x and b is z), all ys (a and b) are related to some z (a is related to a and b, b is related to a)
 The first statement is not true. It is not true that all xs (a and b) are related to all zs (b is not related to a)
A graph is provided in the attached file.

I assume 8 is right
9 I did by using this website http://teachinglogic.liglab.fr/DN/index.php

10 here is what I wrote. It is generic though, we need something specific for 8 a and 9 a.
The tree method starts by assuming the negation of the statement to prove. It then applies a set of rule of derivations in order to find closed branches which provide a countermodel to the negated statement and ultimately prove it wrong (thus proving the original statement).
The natural deduction method instead breaks the statement into an assumption and a consequence and then applies rules which transform the original statement into the cnosequence. In alternative, it can negate the statement and reach a contradiction, thus proving the statement true.